django/db/backends/base/base.py

```diff
@@ -134,7 +134,11 @@ def ensure_connection(self):
     def _cursor(self):
         self.ensure_connection()
         with self.wrap_database_errors:
-            return self.create_cursor()
+            cursor = self.create_cursor()
+            timeout = self.settings_dict['STATEMENT_TIMEOUT']
```

Comment:
Or, if feature is not supported, and timeout is set, raise an exception as well.